### ISSUE TITLE
Parsing error Fortran one line if statements

### ISSUE THREAD

**Issue by mcocdawc** [andymass/vim-matchup#63]
### Explain the issue

1. `NVIM v0.4.0-816-g62d5137c8`
2. Steps to reproduce: MWE
3. Expected behavior: One line if statements (without `then`) should not require a coresponding end if.
4. Observed behavior: One line if statements (without `then`) are parsed, as if they require a coresponding end if.

### Minimal working example

Please provide a minimal working example, e.g.,

```fortran
program matchup_parsing
  real :: x = 1.0, y = 2.0, a


  if (x < y) then
    if (x == 0) stop
    a = y / x
  else
    if (y == 0) stop  ! matchup sees corresponding `if - end if` pairs from HERE
    a = x / y
  end if  ! to HERE

  write(*, *) a

end program
```

### Minimal vimrc file

```vim
call plug#begin('~/.local/share/nvim/plugged')
Plug 'andymass/vim-matchup' 
call plug#end()

filetype plugin indent on
syntax enable
```

**Comment by andymass** [comment]
Thanks!  And sorry, this was introduced in fix for #51 which was clearly invalid.

**Comment by mcocdawc** [comment]
Thank you very much for the fix and the code. ;-)
